cfme/networks/topology.py

```diff
@@ -0,0 +1,222 @@
+import re
+import attr
+
+from navmazing import NavigateToAttribute
+
+from cfme.modeling.base import BaseCollection, BaseEntity
+from cfme.networks.topology_view import TopologyView
+from cfme.utils.appliance.implementations.ui import CFMENavigateStep, navigator, navigate_to
+from wait_for import wait_for
+
+
+@attr.s
+class TopologyLegend(BaseEntity):
+    element = attr.ib()
+    legend_name = attr.ib()
+
+    @property
+    def name(self):
+        return self.legend_name
+
+    @property
+    def is_active(self):
+        return 'active' in self.element.get_attribute('class')
+
+    def set_active(self, active=True):
+        if active != self.is_active:
+            self.element.click()
+
+
+@attr.s
+class TopologyLegendCollection(BaseCollection):
+    """Collection object for legends in topology"""
+    ENTITY = TopologyLegend
+
+    def all(self, topology):
+        final_legends = []
+        legends = topology.browser.elements(topology.view.LEGENDS)
+        for element in legends:
+            legend_name = topology.browser.text(element.find_element_by_tag_name('label'))
+            final_legends.append(self.instantiate(element, legend_name))
+        return final_legends
+
+
+@attr.s
+class TopologyDisplayNames(BaseEntity):
+    element = attr.ib()
+
+    @property
+    def is_enabled(self):
+        return self.element.is_selected()
+
+    def enable(self, enable=True):
+        if self.is_enabled != enable:
+            self.element.click()
+
+    def disable(self):
+        self.enable(enable=False)
+
+
+@attr.s
+class TopologyElement(BaseEntity):
+    element = attr.ib()
+    obj = attr.ib()
+
+    def refresh(self):
+        if self.element is None:
+            raise ValueError('Element should not be None')
+        element_data = re.search('Name: (.*) Type: (.*) Status: (.*)', self.element.text)
+        if len(element_data.groups()) != 3:
+            raise RuntimeError('Topology element does not contain name, type or status')
+        self.name = element_data.group(1)
+        self.type = element_data.group(2)
+        self.status = element_data.group(3)
+        self.x = round(float(self.element.get_attribute('cx')), 1)
+        self.y = round(float(self.element.get_attribute('cy')), 1)
```

Comment:
**REQUIRED**
If this method is used only in `all` of `TopologyElementCollection` you can just extend `instantiate` method.